### PR TITLE
Fix bug of range generator

### PR DESCRIPTION
**Pull request by taiyang-li**
### Changelog category (leave one):
- Bug Fix (user-visible misbehaviour in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
This code segment can prove bug. 
```
int main() 
{
	RangeGenerator g{1230, 100};
	std::cout << g.totalRanges() << std::endl;	
	int count = 0;
	while(g.nextRange())
		++count;
	std::cout << "count:" << count << std::endl;
	return 0;
}

```

### CODE REVIEW

**Comment by antonio2368** [comment]
Good catch.
Maybe add a gtest in `src/Common/tests` to verify for some combinations using a code similar to the example?

**Comment by taiyang-li** [comment]
@antonio2368  ok, I'll try.

**Comment by taiyang-li** [comment]
> Good catch. Maybe add a gtest in `src/Common/tests` to verify for some combinations using a code similar to the example?

So far function `RangeGenerator::totalRanges` is called nowhere. Maybe we don't need to verify it ?

**Comment by antonio2368** [comment]
Well I would either remove it completely or add tests, because it would be problematic if it starts to be used somewhere and it has wrong behaviour.
As adding a small unit test shouldn't be a problem I prefer the option with test.

**Comment by taiyang-li** [comment]
> Well I would either remove it completely or add tests, because it would be problematic if it starts to be used somewhere and it has wrong behaviour. As adding a small unit test shouldn't be a problem I prefer the option with test.

Done.

File: src/Common/tests/gtest_range_generator.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    r = g.nextRange();
+    EXPECT_TRUE(r);
+    EXPECT_EQ(*r, RangeGenerator::Range(10, 20));
+
+    r = g.nextRange();
+    EXPECT_TRUE(r);
+    EXPECT_EQ(*r, RangeGenerator::Range(20, 25));
+
+    r = g.nextRange();
+    EXPECT_TRUE(!r);
```
Comment (antonio2368) [inline src/Common/tests/gtest_range_generator.cpp:25]: This can be simplified with a loop

Comment (taiyang-li) [reply src/Common/tests/gtest_range_generator.cpp]: done

**Review by antonio2368 — APPROVED** [review]

**Comment by taiyang-li** [comment]
Failed tests not related with this pr .

**Pull request merged**